CodeModel/CodeParsers/CSharp/CSharpCodeParser.cs

```diff
@@ -259,23 +260,27 @@ private List<CaDETVariable> LinkMethodVariables(List<CaDETClass> classes, CaDETM
         private List<CaDETClass> GetTypes(List<CaDETClass> classes, CaDETLinkedType type)
         {
             List<CaDETClass> types = new List<CaDETClass>();
-            types.Add(CheckForSingleOrArrayType(classes, type.FullType));
-            types.AddRange(GetCollectionTypes(classes, type.FullType));
+            if (type != null)
+            {
+                types.Add(CheckForSingleOrArrayType(classes, type.FullType));
+                types.AddRange(GetCollectionTypes(classes, type.FullType));
+            }
             return types.Where(t => t != null).ToList();
         }
 
         private CaDETClass CheckForSingleOrArrayType(List<CaDETClass> classes, string fullType)
         {
+            if (fullType == null) return null;
             var foundType = classes.FirstOrDefault(c => c.FullName.Equals(fullType));
             if (foundType != null) return foundType;
-
             foundType = GetArrayType(classes, fullType);
             return foundType;
         }
         
         private List<CaDETClass> GetCollectionTypes(List<CaDETClass> classes, string type)
         {
             List<CaDETClass> collectionTypes = new List<CaDETClass>();
+            if (type == null) return collectionTypes;
```

Comment:
A lot of null checks everywhere. Is it possible to centralize this validation check instead of repeating it in every method?